File: ABC-I/R/init/01_init_02.r
########### - Guia Inicial.
#
# - Objetivo: Primeiro contato com a linguagem R e O RStudio
# - Obs     : Sem Exaurimento.
# -         Funções
# -             * Bult-in
# -             * Not Bult-in
# -         Listas
#
# - Infrastrutura de compilação e informações ao final do Script.
#--------------------------------------------------------------------Thanks.

########### - Parte II - Funções.
#
# - Funcoes
#       - function_name()
# - As funcoes Fazem parte de um pacote
# - Pacotes podem ser bult-in ou baixados posteriormente à intalação da linguagem.
# - Criando uma function 
# - variavel é diferente de funcao pro 
# - Funcao bult-in c - o que ela faz?
c <- c(a,b)
c
# help da funcao
?c
c <- c("logica", "programacao")
c

a <- c(23,32,12,56)
a
# - como a funcao c vetoriza, ou seja, armazena objetos em posiçoes especificas.
# - podemos acessá-los.
a[0]
c[0]

a[2]
# - outro exemplo de funcao bult-in
?summary
summary(a)
summary(c)

#_utilizando uma funcao não bult-in, ou seja, que não vem por padrao
?str_c

install.packages("stringr") # intala o pacote stringr que contem as funçoes abaixo.
?str_c
??str_c

#perceba que, por não ser um package bult-in, need import
library("stringr")
?str_c
a <- "first name"
b <- "Second"
d <- str_c(a,b)
f <- c(a,b)
#veja a diferença.
d
f
########### --------------------------------------------------------------------Reloaded.

########### - Compile_By.
#
# - SO:  Windows 10 Pro.22H2
# - IDE: RStudio.2023.09.0
# - Obs: no bugs.
# - Obs: 01/10/2023.
#
# - Hello... please if you find something wrong,  contact-me.
#--------------------------------------------------------------------Follow the white rabbit.

File: ABC-I/C++/01_init_02.cpp
//########### - Guia Inicial.
//#
//# - Objetivo: Primeiro contato com a linguagem C++ e p DEV C++
//# - Obs     : Sem Exaurimento.
//# -     * Topicos Abordados
//# -         Variáveis
//# -             Criação
//# -             Atribuição
//# -             C++ Case Sensitive
//# -             Strings
//#
//# - Infrastrutura de compilação e informações ao final do Script.
//########### --------------------------------------------------------------------Thanks.

//########### - Parte I - inclusao das bibliotecas.
#include <iostream>
    using namespace std;

//#main function
int main(){
	printf("Hello.\n");
    //Variables
    //TIPO NOME;
    //TIPO NOME = VALOR;

    //Declare and initialize
    int vidas=0; //10, 25
    char letra='A'; //'B'
    double Real=5.2; //3.143982341234341234
    float Real_=5.2; //3.14
    bool logic=true;  //TRUE or FALSE
    string nome=".\H.Meneses"; //"list of char"
    char vector[20]; //"20 position same type"

    //É interessante buscar exaurir leitura sobre type of variables

    //Showing with cout
    cout << vidas << "\n";
    cout << letra << "\n";
    cout << Real  << "\n";
    cout << Real_ << "\n";
    cout << logic << "\n";
    cout << nome  << "\n";

    //Another way
    cout << vidas << "\n" << letra << "\n";
    cout << Real  << "\n" << Real_ << "\n";
    cout << logic << "\n" << nome  << "\n";
    
    cout << "Digite o valor de vidas: ";
    cin >> vidas;
    cout << "Digite o valor seu nome: ";
    cin >> nome;
    cout << "Apresntando os valores modificados: ";
    cout << vidas << "\n" << nome << "\n";

    //mandando comando para o sistema.
    system("pause");

    return 0;
}


//########### --------------------------------------------------------------------Reloaded.


//########### - Compile_By.
//#
//# - SO:  Windows 10 Pro.22H2
//# - IDE: DEV C++ Version 5.11
//# - Obs: no bugs.
//# - Obs: 17/10/2023.
//#
//# - Hello... please if you find something wrong,  contact-me.
//########### --------------------------------------------------------------------Follow the white rabbit.
